cnd/src/lnd.rs

```diff
@@ -57,27 +57,68 @@ struct Payment {
 pub struct LndConnectorParams {
     pub lnd_url: Url,
     pub retry_interval_ms: u64,
-    pub certificate: Certificate,
+    pub certificate_path: PathBuf,
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
+enum LazyCertificate {
+    Path(PathBuf),
+    Certificate(Certificate),
+}
+
+impl LazyCertificate {
+    pub fn new(path: PathBuf) -> Self {
+        Self::Path(path)
+    }
+
+    pub fn read(self) -> Result<Self, Error> {
+        use LazyCertificate::*;
+
+        match self {
+            Certificate(_) => Ok(self),
+            Path(path) => {
+                let mut buf = Vec::new();
+                std::fs::File::open(path)?.read_to_end(&mut buf)?;
+                let certificate = reqwest::Certificate::from_pem(&buf)?;
+                Ok(LazyCertificate::Certificate(certificate))
+            }
+        }
+    }
+
+    pub fn certificate(&self) -> Result<&Certificate, Error> {
```

Comment:
@D4nte 

Instead of erroring here, you could just do the read logic then you don't have to care about the error case :)

Reply:
Do you mean read the file in this function?

Yes it is what I have done initially however, I did not want to read the file "every time".
To not have to read the file everytime I have to have a `&mut self` and it just bubbled up too much.

Hence the current solution where I do not read the current file has often.